spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -318,9 +332,30 @@ protected void doStop() {
 
 		private final AckMode ackMode = getAckMode();
 
+		private final boolean isManualAck = this.ackMode.equals(AckMode.MANUAL);
+
+		private final boolean isManualImmediateAck = this.ackMode.equals(AckMode.MANUAL_IMMEDIATE)
+				|| this.ackMode.equals(AckMode.MANUAL_IMMEDIATE_SYNC);
+
+		private final boolean isAnyManualAck = this.isManualAck || this.isManualImmediateAck;
+
+		private final boolean isRecordAck = this.ackMode.equals(AckMode.RECORD);
+
 		private final boolean syncCommits = KafkaMessageListenerContainer.this.syncCommits;
 
-		private Thread consumerThread;
+		private final long pauseAfter = getPauseAfter();
+
+		private final boolean pauseEnabled = isPauseEnabled();
+
+		private final BlockingQueue<ConsumerRecords<K, V>> recordsToProcess = new LinkedBlockingQueue<>(getQueueDepth());
+
+		private final int queueDepth = getQueueDepth();
+
+		private final BlockingQueue<ConsumerRecord<K, V>> acks = new LinkedBlockingQueue<>();
```

Comment:
So it seems that on revoke, we should suspend the listener thread, commit any uncommitted acks, wait for the new assignment, restart the listener, which should then skip any queued messages that are for partitions that are no longer assigned to us and process any that are.